graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt

```diff
@@ -43,6 +43,7 @@ import kotlin.reflect.full.findAnnotation
  */
 open class FederatedSchemaGeneratorHooks(private val federatedTypeRegistry: FederatedTypeRegistry) : SchemaGeneratorHooks {
     private val directiveRegex = "(^#.+$[\\r\\n])?^directive @\\w+.+$[\\r\\n]*".toRegex(setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
+    private val declaredDirectiveRegex = "@(?!extends)(?!external)(?!key)(?!provides)(?!requires)\\w+(?:\\(.*(?=\\))\\))?".toRegex(setOf(RegexOption.MULTILINE, RegexOption.IGNORE_CASE))
```

Comment:
can you also update the comment here -> https://github.com/ExpediaGroup/graphql-kotlin/blob/7f5e13f656ad4764184ccd42688c4ccdbbba6793/graphql-kotlin-federation/src/main/kotlin/com/expediagroup/graphql/federation/FederatedSchemaGeneratorHooks.kt#L83 to 

```
// SDL returned by _service query should NOT contain
// - default schema definition
// - empty Query type
// - any directive definitions
// - any custom directives
// - new federated scalars
```